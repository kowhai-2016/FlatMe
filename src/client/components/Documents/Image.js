import React, { PropTypes } from 'react'

const Image = props => {
  return <img src={props.image.url} />
}

Image.propTypes = {
  image: PropTypes.shape({
    url: PropTypes.string.isRequired
  }).isRequired
}

export default Image
