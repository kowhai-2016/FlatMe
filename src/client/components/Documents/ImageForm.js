import React, { PropTypes } from 'react'

const ImageForm = React.createClass({
  componentDidMount () {
    const flatId = this.props.flatId
    function uploadFile (file, signedRequest, url) {
      const xhr = new window.XMLHttpRequest()
      xhr.open('PUT', signedRequest)
      xhr.onreadystatechange = () => {
        if (xhr.readyState === 4) {
          if (xhr.status === 200) {
            document.getElementById('preview').src = url
            document.getElementById('avatar-url').value = url
          } else {
            window.alert('Could not upload file.')
          }
        }
      }
      xhr.send(file)
    }

    function getSignedRequest (file) {
      const xhr = new window.XMLHttpRequest()
      xhr.open('GET', `/v1/flats/${flatId}/documents/sign-s3?file-name=${file.name}&file-type=${file.type}`)
      xhr.onreadystatechange = () => {
        if (xhr.readyState === 4) {
          if (xhr.status === 200) {
            const response = JSON.parse(xhr.responseText)
            uploadFile(file, response.signedRequest, response.url)
          } else {
            window.alert('Could not get signed URL.')
          }
        }
      }
      xhr.send()
    }

    document.getElementById('file-input').onchange = () => {
      const files = document.getElementById('file-input').files
      const file = files[0]
      if (file == null) {
        window.alert('No file selected.')
      }
      getSignedRequest(file)
    }
  },
  render () {
    return (
      <div>
        <input type='file' id='file-input' />
        <p id='status'>Please select a file</p>
        <img id='preview' />
        <form method='POST' action={`/v1/flats/${this.props.flatId}/documents`}>
          <input type='hidden' id='avatar-url' name='avatar-url' value='/images/default.jpg' />
          <input type='submit' value='Update profile' />
        </form>
      </div>
    )
  }
})

ImageForm.propTypes = {
  flatId: PropTypes.number.isRequired
}

export default ImageForm
