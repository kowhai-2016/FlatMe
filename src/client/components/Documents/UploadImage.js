import React, { PropTypes } from 'react'

const UploadImage = props => {
  return (
    <form
      action={`/v1/flats/${props.flatId}/documents`}
      encType='multipart/form-data'
      method='post'>
      <input name='image' type='file' />
      <input type='submit' />
    </form>
  )
}

UploadImage.propTypes = {
  flatId: PropTypes.number.isRequired
}

export default UploadImage
