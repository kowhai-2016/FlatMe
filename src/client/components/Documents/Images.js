import React, { PropTypes } from 'react'

import Image from './Image'

const Images = props => {
  return (
    <div>
      {props.images.map((image, key) => <Image key={key} image={image} />)}
    </div>
  )
}

Images.propTypes = {
  images: PropTypes.arrayOf(
    PropTypes.shape({
      url: PropTypes.string.isRequired
    }).isRequired
  )
}

export default Images
