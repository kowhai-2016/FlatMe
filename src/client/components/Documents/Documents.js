import React, { PropTypes } from 'react'

import ImageForm from './ImageForm'

const Documents = React.createClass({
  render () {
    return (
      <div>
        <h1>Documents</h1>
        {this.props.flat ? <ImageForm flatId={this.props.flat.id} /> : null}
      </div>
    )
  }
})

Documents.propTypes = {
  flat: PropTypes.object
}

export default Documents
