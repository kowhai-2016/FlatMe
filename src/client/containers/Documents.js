import { connect } from 'react-redux'

import Documents from '../components/Documents'

const mapStateToProps = (state, ownProps) => {
  const id = Number(ownProps.params.id)
  const flat = state.flats[id]
  let documents = []
  if (flat && flat.documents) {
    documents = flat.documents
  }
  return {
    documents
  }
}

export default connect(mapStateToProps)(Documents)
