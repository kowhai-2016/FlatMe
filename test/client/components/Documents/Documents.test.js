/* eslint-env jest */

import React from 'react'
import renderer from 'react-test-renderer'

import Documents from '../../../../src/client/components/Documents'

test('Documents renders correctly', () => {
  const documents = []
  const params = {
    id: '1'
  }
  const tree = renderer.create(
    <Documents documents={documents} params={params} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
