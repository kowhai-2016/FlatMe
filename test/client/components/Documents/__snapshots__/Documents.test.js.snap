exports[`test Documents renders correctly 1`] = `
<div>
  <h1>
    Documents
  </h1>
</div>
`;
